### ISSUE TITLE
using linq2db.MySqlConnector , but error like using "SqlServer"

### ISSUE THREAD

**Issue by ifew** [linq2db/linq2db#1715]
I using  ```linq2db.MySqlConnector``` and develop follow this link ```https://linq2db.github.io/#now-lets-create-a-poco-class``` 

but I got an error message not relate "MySql", it's look like "SqlServer"

This error message

```bash
Payload:
{
  "errorType": "AggregateException",
  "errorMessage": "One or more errors occurred. (The type initializer for 'LinqToDB.Data.DataConnection' threw an exception.)",
  "stackTrace": [
    "at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)",
    "at lambda_method(Closure , Stream , Stream , LambdaContextInternal )"
  ],
  "cause":   {
    "errorType": "TypeInitializationException",
    "errorMessage": "The type initializer for 'LinqToDB.Data.DataConnection' threw an exception.",
    "stackTrace": [
      "at unit_detail.Services..ctor() in /Users/chitpong/Project/project/sourcecode_orm/project-api/project-api/src/unit_detail_noef/Services.cs:line 24",
      "--- End of stack trace from previous location where exception was thrown ---",
      "at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCreateInstance(CreateInstanceCallSite createInstanceCallSite, ServiceProviderEngineScope scope)",
      "at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScoped(ScopedCallSite scopedCallSite, ServiceProviderEngineScope scope)",
      "at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetService[T](IServiceProvider provider)",
      "at unit_detail.Function.FunctionHandler(APIGatewayProxyRequest request, ILambdaContext context) in /Users/chitpong/Project/project/sourcecode_orm/project-api/project-api/src/unit_detail_noef/Function.cs:line 68"
    ],
    "cause":     {
      "errorType": "TypeInitializationException",
      "errorMessage": "The type initializer for 'LinqToDB.DataProvider.SqlServer.SqlServerTools' threw an exception.",
      "stackTrace": [
        "at LinqToDB.DataProvider.SqlServer.SqlServerTools.GetDataProvider(SqlServerVersion version)",
        "at LinqToDB.Data.DataConnection..cctor()"
      ],
      "cause":       {
        "errorType": "FileNotFoundException",
        "errorMessage": "Could not load file or assembly 'System.Data.SqlClient, Version=4.5.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.\n",
        "stackTrace": [
          "at LinqToDB.DataProvider.SqlServer.SqlServerDataProvider..ctor(String name, SqlServerVersion version)",
          "at LinqToDB.DataProvider.SqlServer.SqlServerTools..cctor()"
        ]
      }
    }
  },
  "causes": [    {
      "errorType": "TypeInitializationException",
      "errorMessage": "The type initializer for 'LinqToDB.Data.DataConnection' threw an exception.",
      "stackTrace": [
        "at unit_detail.Services..ctor() in /Users/chitpong/Project/project/sourcecode_orm/project-api/project-api/src/unit_detail_noef/Services.cs:line 24",
        "--- End of stack trace from previous location where exception was thrown ---",
        "at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCreateInstance(CreateInstanceCallSite createInstanceCallSite, ServiceProviderEngineScope scope)",
        "at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScoped(ScopedCallSite scopedCallSite, ServiceProviderEngineScope scope)",
        "at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetService[T](IServiceProvider provider)",
        "at unit_detail.Function.FunctionHandler(APIGatewayProxyRequest request, ILambdaContext context) in /Users/chitpong/Project/project/sourcecode_orm/project-api/project-api/src/unit_detail_noef/Function.cs:line 68"
      ],
      "cause":       {
        "errorType": "TypeInitializationException",
        "errorMessage": "The type initializer for 'LinqToDB.DataProvider.SqlServer.SqlServerTools' threw an exception.",
        "stackTrace": [
          "at LinqToDB.DataProvider.SqlServer.SqlServerTools.GetDataProvider(SqlServerVersion version)",
          "at LinqToDB.Data.DataConnection..cctor()"
        ],
        "cause":         {
          "errorType": "FileNotFoundException",
          "errorMessage": "Could not load file or assembly 'System.Data.SqlClient, Version=4.5.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.\n",
          "stackTrace": [
            "at LinqToDB.DataProvider.SqlServer.SqlServerDataProvider..ctor(String name, SqlServerVersion version)",
            "at LinqToDB.DataProvider.SqlServer.SqlServerTools..cctor()"
          ]
        }
      }
    }
  ]
}


Log Tail:
START RequestId: 033cc114-1706-4649-8ae3-6577ca646b54 Version: $LATEST
One or more errors occurred. (The type initializer for 'LinqToDB.Data.DataConnection' threw an exception.): AggregateException
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at lambda_method(Closure , Stream , Stream , LambdaContextInternal )

   at unit_detail.Services..ctor() in /Users/chitpong/Project/project/sourcecode_orm/project-api/project-api/src/unit_detail_noef/Services.cs:line 24
   --- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCreateInstance(CreateInstanceCallSite createInstanceCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScoped(ScopedCallSite scopedCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetService[T](IServiceProvider provider)
   at unit_detail.Function.FunctionHandler(APIGatewayProxyRequest request, ILambdaContext context) in /Users/chitpong/Project/project/sourcecode_orm/project-api/project-api/src/unit_detail_noef/Function.cs:line 68
The type initializer for 'LinqToDB.DataProvider.SqlServer.SqlServerTools' threw an exception.: TypeInitializationException
   at LinqToDB.DataProvider.SqlServer.SqlServerTools.GetDataProvider(SqlServerVersion version)
   at LinqToDB.Data.DataConnection..cctor()
Could not load file or assembly 'System.Data.SqlClient, Version=4.5.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.: FileNotFoundException
   at LinqToDB.DataProvider.SqlServer.SqlServerDataProvider..ctor(String name, SqlServerVersion version)
   at LinqToDB.DataProvider.SqlServer.SqlServerTools..cctor()
```

This my code

```cs
public class ConnectionStringSettings : IConnectionStringSettings
    {
        public string ConnectionString { get; set; }
        public string Name { get; set; }
        public string ProviderName { get; set; }
        public bool IsGlobal => false;
    }

    public class MySettings : ILinqToDBSettings
    {
        public IEnumerable<IDataProviderSettings> DataProviders => Enumerable.Empty<IDataProviderSettings>();

        public string DefaultConfiguration => "MyDatabase";
        public string DefaultDataProvider => "MySqlConnector";

        public IEnumerable<IConnectionStringSettings> ConnectionStrings
        {
            get
            {
                yield return
                    new ConnectionStringSettings
                    {
                        Name = "MyDatabase",
                        ProviderName = "MySqlConnector",
                        ConnectionString = "server=xxxx.com;userid=xxxxx;password=xxxxx;database=xxxxx;convert zero datetime=True; CharSet=utf8;"
                    };
            }
        }
    }
```

.Net Core 2.1
linq2db.MySqlConnector 2.7.0

**Comment by sdanyliv** [comment]
Sorry not near computer. I know it may look weird but try to install System.Data.SqlClient library 4.5.1 or latest in your project.

**Comment by ifew** [comment]
> Sorry not near computer. I know it may look weird but try to install System.Data.SqlClient library 4.5.1 or latest in your project.

Hi @sdanyliv , It's work!

Why I need to using "System.Data.SqlClient"?

**Comment by sdanyliv** [comment]
@ifew, it is something from .NET framework legacy. Library in development more than 10 years and some providers needs improvements. We have issue for that and i hope in next minor release we will remove this dependency.

**Comment by ifew** [comment]
> @ifew, it is something from .NET framework legacy. Library in development more than 10 years and some providers needs improvements. We have issue for that and i hope in next minor release we will remove this dependency.

@sdanyliv Thank you guy, I have asked because of implement for AWS Lambda and it has affected with cold start duration time for any kind of library to used

**Comment by MaceWindu** [comment]
Duplicate of #1704

**Comment by MaceWindu** [comment]
I'm reopening it, as a fix for #1704 will be just a quick fix to dependency versions. We will use this issue to track removal of explicit system.data.sqlclient dependency

**Comment by MaceWindu** [comment]
fixed by #1929
